src/oic/utils/keyio.py

```diff
@@ -776,6 +776,36 @@ def get_issuer_keys(self, issuer):
             res.extend(kbl.keys())
         return res
 
+    def __eq__(self, other):
+        """
+        Compare 2 KeyJar instances
+        :param other: The other KeyJar instance
+        :return: True/False
+        """
+        try:
+            assert isinstance(other, KeyJar)
+        except AssertionError:
+            return False
+
+        # number of issuer_keys must be equal
+        if self.issuer_keys.keys() != other.issuer_keys.keys():
+            return False
+
+        for issuer in self.issuer_keys.keys():
+            mine = self.get_issuer_keys(issuer)
+            others = other.get_issuer_keys(issuer)
+            if len(mine) != len(others):
+                return False
+            for a_key in mine:
```

Comment:
I think you could replace this nested for logic as `return set(mine) & set(others)`, no?

https://docs.python.org/3/library/stdtypes.html#set